cli.go

```diff
@@ -74,6 +75,12 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
 		callbackPort = port
 	}
 
+	doBrowserLaunch := defaultBrowserLaunch
+	_, ok = m["no-launch"]
```

Comment:
I'm thinking we should check the value here, since we're documenting this option as a boolean. Otherwise, the no-launch behavior will still occur if you provide the default `no-launch=false` to the CLI.

Reply:
This `Auth` method is called from [login.go#L225](https://github.com/hashicorp/vault/blob/main/command/login.go#L225). See that there is an [interface](https://github.com/hashicorp/vault/blob/main/command/login.go#L13-L18) to implement which restricts this to using `map[string]string`. 

There is precedent for parsing string values as booleans using [strconv#ParseBool](https://pkg.go.dev/strconv#ParseBool), so I suggest we take that approach. Here is an example in Vault: [token/cli.go#L27](https://github.com/hashicorp/vault/blob/main/builtin/credential/token/cli.go#L27). Thanks 👍

Reply:
Alright, made the update. Pending whatever changes get decided in the actual name.

Aside: I wanted to add a compile-time check for the interface, but that creates an import loop. On hindsight that's obvious. Is command.LoginHandler the only exported/pluggable interface in vault's cli?

Reply:
Thanks! LGTM. That's a good idea on the compile-time check, but you're right about the import loop. As far as I know, command.LoginHandler is the only interface for plugins to use for CLI handling.